opentelemetry-instrumentation/src/opentelemetry/instrumentation/propagators.py

```diff
@@ -0,0 +1,108 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This module implements experimental propagators to inject trace response context
```

Comment:
A separate branch for anything experimental is too big a barrier IMO for trying out things. This package (instrumentation) is already experimental or unstable since it's pre-1.0. I don't see a lot of value in doing this. 

This is basically just some common code that a lot of instrumentations share. It's more like what we generally call "util" code but it looks like propagators in API so we documented it as experimental so users don't start relying on it. It was kind of redundant as the package it is in is already unstable and subject to change. 

If this was in API package, we could have had a different branch for it or put it under a submodule like `opentelemetry.api.experimental.` but don't see much benefit in doing that if we store it in instrumentation. I can still put it under an `experimental` submodule within instrumentation but the entire package is experimental in a sense already so not sure if it helps.

Reply:
We are also planning to move `opentelemetry-instrumentation` to contrib as discussed way before.